llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -3261,6 +3261,20 @@ Instruction *InstCombinerImpl::foldICmpBitCast(ICmpInst &Cmp) {
                                 ConstantInt::getAllOnesValue(NewType));
         }
       }
+
+      // icmp eq/ne (bitcast X to int), special fp -> llvm.is.fpclass(X, class)
+      Type *FPType = SrcType->getScalarType();
+      if (!Cmp.getParent()->getParent()->hasFnAttribute(
+              Attribute::NoImplicitFloat) &&
+          Cmp.isEquality() && FPType->isIEEELikeFPTy()) {
+        unsigned Mask = APFloat(FPType->getFltSemantics(), *C).classify();
```

Comment:
I believe classify returns the true FPClassTest, you don't need the implicit cast to unsigned

Reply:
Done.